onboard/rocks_catalog_actions.go

```diff
@@ -95,12 +115,12 @@ func (c *RocksCatalogRepoActions) Commit(ctx context.Context, commitMsg string,
 	if c.createdMetaRangeID == nil {
 		return "", ErrNoMetaRange
 	}
-	commitID, err := c.metaRanger.CommitExistingMetaRange(ctx, c.repoID, c.commitID, *c.createdMetaRangeID, c.committer, commitMsg, graveler.Metadata(metadata))
+	commitID, err := c.entryCataloger.CommitExistingMetaRange(ctx, c.repoID, c.previousCommitID, *c.createdMetaRangeID, c.committer, commitMsg, graveler.Metadata(metadata))
```

Comment:
I think we have a concurrency problem here. It is not created by this PR, but becomes more problematic when using master.

Suppose during the import, someone performs commit A to master.
We started the import when there were no commits to master.
So we now create a commit B which has no parent, and we set master to point to it. We detached commit A from our repository. I don't have a real solution for that. Maybe the relevant branch should be locked throughout the entire process?

Reply:
I agree, and I think we discussed that during the previous PR of import.
We don't provide any graveler-external locking for now, this PR is not really the place to introduce one. 

I'm opening an issue to track this, I think we probably have a similar issue with `migrate` #1313 .

The real problem is with 2 detached commits. Anything else is still a race, but one you can recover from easily. Up until now, import did not create the first commit to the repo.